lib/quadquizaminos_web/live/tetris_live.ex

```diff
@@ -619,19 +620,34 @@ defmodule QuadquizaminosWeb.TetrisLive do
   end
 
   defp move_block(socket, x, y, block_coordinates, true = _adding_block, true = _moving_block) do
-    {x, y} = parse_to_integer(x, y)
-    {x1, y1, color} = block_coordinates
-    powers = socket.assigns.powers -- [:moveblock]
-
-    bottom =
-      socket.assigns.bottom |> Map.delete({x1, y1}) |> Map.merge(%{{x, y} => {x, y, color}})
+    # check if the coordinates are part of bottom, if they're not return the socket, if they're apply transformation
+    # and return socket
+    ycoordinate =
+      socket.assigns.bottom
+      |> Map.keys()
+      |> Enum.map(fn value -> elem(value, 1) end)
+      |> Enum.map(fn c -> Integer.to_string(c) end)
+
+    if y in ycoordinate do
+      {x, y} = parse_to_integer(x, y)
+      {x1, y1, color} = block_coordinates
 
-    assign(socket,
-      bottom: bottom,
-      powers: powers,
-      adding_block: false,
-      moving_block: false
-    )
+      bottom =
+        socket.assigns.bottom
+        |> Map.delete({x1, y1})
+        |> Map.merge(%{{x, y} => {x, y, color}})
+
+      powers = socket.assigns.powers -- [:moveblock]
+      
+      assign(socket,
+        bottom: bottom,
+        powers: powers,
+        adding_block: false,
+        moving_block: false
+      )
+    else
+      assign(socket, powers: powers, moving_block: false, adding_block: false)
```

Comment:
does that mean If we reset `moving_block` and `adding_block` to `false` we will be able to click on the super power again and use it?

Reply:
yes, the two events, have a hovering block so setting them to false does away with that but the power-up remains, so when the brick becomes part of the bottom, then a player can use the power-up previously earned at that point